### PR TITLE
ENH: Implement IntegerArray.sum

### PR DESCRIPTION
**Pull request by dsaxton**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

I think this is mostly interesting in that it allows normalize=True for value_counts on an IntegerArray backed Series, which currently doesn't work:

```python
[ins] In [1]: s = pd.Series([1, 2, 3], dtype="Int64")

[ins] In [2]: s.value_counts(normalize=True)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-2bf1a78353e5> in <module>
----> 1 s.value_counts(normalize=True)

~/pandas/pandas/core/base.py in value_counts(self, normalize, sort, ascending, bins, dropna)
   1252             normalize=normalize,
   1253             bins=bins,
-> 1254             dropna=dropna,
   1255         )
   1256         return result

~/pandas/pandas/core/algorithms.py in value_counts(values, sort, ascending, normalize, bins, dropna)
    725
    726     if normalize:
--> 727         result = result / float(counts.sum())
    728
    729     return result

AttributeError: 'IntegerArray' object has no attribute 'sum'
```

### CODE REVIEW

File: pandas/core/arrays/integer.py [hunk 1]
```diff
@@ -573,6 +573,12 @@ def _reduce(self, name: str, skipna: bool = True, **kwargs):
 
         return result
 
+    def sum(self, skipna: bool = True, min_count: int = 0):
```
Comment (jbrockmendel) [inline pandas/core/arrays/integer.py:576]: probably should make the signature match PandasArray.sum etc

File: pandas/tests/arrays/integer/test_function.py [hunk 2]
```diff
@@ -113,6 +113,17 @@ def test_value_counts_empty():
     tm.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize("skipna", [True, False])
+@pytest.mark.parametrize("min_count", [0, 4])
+def test_integer_array_sum(skipna, min_count):
```
Comment (jbrockmendel) [inline pandas/tests/arrays/integer/test_function.py:118]: is e.g. Series[Int64].sum() or DataFrame[Int64].sum() fixed by this?

File: pandas/core/arrays/integer.py (hunk 1 above)
Comment (dsaxton) [reply pandas/core/arrays/integer.py:576]: Ok, and we still want to keep axis (even though it wouldn't be used here)?

File: pandas/tests/arrays/integer/test_function.py (hunk 2 above)
Comment (dsaxton) [reply pandas/tests/arrays/integer/test_function.py:118]: Those are actually already working, just not for IntegerArray specifically

**Review by jreback — CHANGES_REQUESTED** [review]
this is not the correct way to fix this. series/frames *already* dispatch to EAs via _reduce, which DO implement sum/prod/min/max. The issue is these functions need to be exposed via that dispatch mechanism, rather than writing more code to actually do the reduction.

**Comment by jreback** [comment]
furthermore the numpy_ EA do this in an opposite way, meaning that the ops themselves are defined, and _reduce dispatches TO them.

So we should decide on a single way forward here. I think the way integer array does this is better, e.g. in _reduce.

File: doc/source/whatsnew/v1.1.0.rst [hunk 3]
```diff
@@ -601,6 +601,7 @@ Other
 - Bug in :meth:`Series.map` not raising on invalid ``na_action`` (:issue:`32815`)
 - Bug in :meth:`DataFrame.__dir__` caused a segfault when using unicode surrogates in a column name (:issue:`25509`)
 - Bug in :meth:`DataFrame.plot.scatter` caused an error when plotting variable marker sizes (:issue:`32904`)
+- :class:`IntegerArray` now implements the ``sum`` operation (:issue:`33538`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.1.0.rst:604]: since this PR would fix #33172. maybe the release not could reflect this.

**Comment by jorisvandenbossche** [comment]
As mentioned in https://github.com/pandas-dev/pandas/pull/33351#issuecomment-610868353, we should probably have a general discussion about how to deal with those reduction methods in our EAs, otherwise those comments like Jeff's concern above (https://github.com/pandas-dev/pandas/pull/33538) are going to keep coming up in each PR that adds something like this.

File: pandas/core/arrays/integer.py [hunk 4]
```diff
@@ -573,6 +574,24 @@ def _reduce(self, name: str, skipna: bool = True, **kwargs):
 
         return result
 
+    def sum(
+        self,
+        axis=None,
+        dtype=None,
+        out=None,
+        keepdims=False,
+        initial=None,
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/integer.py:583]: Can you remove all those unnecessary keywords and put them in a **kwargs ?

**Review by jreback — APPROVED** [review]
lgtm. @jorisvandenbossche agree we agree on the issues you mentioned, but happy with this for now.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
thanks @dsaxton !

**Comment by jorisvandenbossche** [comment]
@dsaxton would you like to open a general issue about this? (https://github.com/pandas-dev/pandas/pull/33538#issuecomment-618949767)

**Pull request merged**